apps/forklift/test/unit/integration/event_handling_test.exs

```diff
@@ -85,4 +87,24 @@ defmodule Forklift.Integration.EventHandlingTest do
       assert_receive %SmartCity.Dataset{id: "terminate-id"}
     end
   end
+
+  test "should delete dataset when dataset:delete event handle is called" do
+    dataset = TDG.create_dataset(id: Faker.UUID.v4(), technical: %{sourceType: "ingest"})
+    expect(MockReader, :terminate, fn _ -> :ok end)
+    expect(MockTopic, :delete, fn _ -> :ok end)
+    expect(MockTable, :delete, fn _ -> :ok end)
+    expect(Forklift.Datasets.delete(any()), return: :ok)
+
+    Brook.Test.with_event(instance_name(), fn ->
+      Forklift.EventHandler.handle_event(
+        Brook.Event.new(
+          type: dataset_delete(),
+          data: dataset,
+          author: :author
+        )
+      )
+    end)
+
+    assert_called(Forklift.Datasets.delete(dataset.id))
```

Comment:
Is this assertion somewhat redundant since you have the expect on line 96?  I think you could just change your expect to `expect(Forklift.Datasets.delete(dataset.id), return: :ok)`

Reply:
You can remove this assertion now.